src/controllers/controller.bar.js

```diff
@@ -267,50 +273,53 @@ module.exports = DatasetController.extend({
 		meta.bar = true;
 	},
 
-	update: function(reset) {
+	update: function(mode) {
 		const me = this;
 		const rects = me._cachedMeta.data;
 
-		me.updateElements(rects, 0, rects.length, reset);
+		me.updateElements(rects, 0, rects.length, mode);
 	},
 
-	updateElements: function(rectangles, start, count, reset) {
+	updateElements: function(rectangles, start, count, mode) {
 		const me = this;
+		const reset = mode === 'reset';
 		const vscale = me._cachedMeta.vScale;
 		const base = vscale.getBasePixel();
 		const horizontal = vscale.isHorizontal();
 		const ruler = me.getRuler();
+		const firstOpts = me._resolveDataElementOptions(start, mode);
+		const sharedOptions = me._getSharedOptions(mode, rectangles[start], firstOpts);
+		const includeOptions = me._includeOptions(mode, sharedOptions);
+
 		let i;
 
 		for (i = 0; i < start + count; i++) {
-			const rectangle = rectangles[i];
-			const options = me._resolveDataElementOptions(i);
+			const options = me._resolveDataElementOptions(i, mode);
 			const vpixels = me.calculateBarValuePixels(i, options);
 			const ipixels = me.calculateBarIndexPixels(i, ruler, options);
 
-			rectangle._model = {
-				backgroundColor: options.backgroundColor,
-				borderColor: options.borderColor,
-				borderSkipped: options.borderSkipped,
-				borderWidth: options.borderWidth
+			const properties = {
+				horizontal,
+				base: reset ? base : vpixels.base,
+				x: horizontal ? reset ? base : vpixels.head : ipixels.center,
+				y: horizontal ? ipixels.center : reset ? base : vpixels.head,
+				height: horizontal ? ipixels.size : undefined,
+				width: horizontal ? undefined : ipixels.size
 			};
 
-			const model = rectangle._model;
-
 			// all borders are drawn for floating bar
+			/* TODO: float bars border skipping magic
```

Comment:
I kind of don't like this kind of things in the code. We don't really know if the user wants to draw all borders in the float bar. It could be default though, but that's kind of hard because float bar does not have its own type.

So this could be "fixed" in many ways and has little to do with the animations.
- it could be just documented, so users can add the config
- we could separate floatBar to its own type and provide defaults
- we could continue overwriting any user config here (not my favorite)
  * I think I did not do that because those "shared" properties were frozen initially and this caused problems. I ended up removing the freezing though.